scan-command/src/main/java/io/ballerina/scan/internal/StaticCodeAnalyzer.java

```diff
@@ -100,6 +116,77 @@ public void visit(ImplicitAnonymousFunctionExpressionNode implicitAnonymousFunct
         this.visitSyntaxNode(implicitAnonymousFunctionExpressionNode.expression());
     }
 
+    @Override
+    public void visit(ClassDefinitionNode classDefinitionNode) {
+        checkNonIsolatedPublicClassDefinitions(classDefinitionNode);
+        checkNonIsolatedPublicClassMembers(classDefinitionNode);
+    }
+
+    private void checkNonIsolatedPublicClassDefinitions(ClassDefinitionNode classDefinitionNode) {
+        semanticModel.symbol(classDefinitionNode).ifPresent(symbol -> {
+            if (symbol instanceof ClassSymbol classSymbol) {
+                List<Qualifier> qualifiers = classSymbol.qualifiers();
+                if (getQualifier(qualifiers, PUBLIC_KEYWORD.stringValue()) &&
+                        !getQualifier(qualifiers, ISOLATED_KEYWORD.stringValue())) {
+                    reportIssue(classDefinitionNode, CoreRule.PUBLIC_NON_ISOLATED_CONSTRUCT);
+                }
+            }
+        });
+    }
+
+    private void checkNonIsolatedPublicClassMembers(ClassDefinitionNode classDefinitionNode) {
+        classDefinitionNode.members().forEach(member -> {
+            semanticModel.symbol(member).ifPresent(symbol -> {
+                if (symbol.kind() == SymbolKind.METHOD) {
```

Comment:
Previously this `if (symbol instanceof ClassSymbol classSymbol) {` instanceof check was used to cast the symbol we do the same thing here and be consistent. WDYT?

Reply:
In there we use the value after casting. But here we just need to check the kind